Modules/_elementtree.c

```diff
@@ -994,6 +994,10 @@ element_setstate_from_attributes(ElementObject *self,
     }
     assert(self->extra && self->extra->allocated >= nchildren);
 
+    /* Decref each of the old children. */
+    for (i = 0; i < self->extra->length; i++) {
+        Py_DECREF(self->extra->children[i]);
```

Comment:
`Py_DECREF()` can cause executing arbitrary Python code. This can change the size of the `children` list. `PyList_GET_ITEM()` in the following loop can crash.

Reply:
Ah, indeed.
Then ISTM that we should try to write a helper function that would be used both here and by `element_ass_subscr()`.
what do you think?